ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -75,12 +85,22 @@ public DdlCommandResult executeCreateStream(final CreateStreamCommand createStre
           withQuery,
           getKsqlTopic(createStream)
       );
+
       metaStore.putSource(ksqlStream, createStream.isOrReplace());
       return new DdlCommandResult(true, "Stream created");
     }
 
     @Override
     public DdlCommandResult executeCreateTable(final CreateTableCommand createTable) {
+      final SourceName sourceName = createTable.getSourceName();
+      final DataSource dataSource = metaStore.getSource(sourceName);
+
+      if (dataSource != null && !createTable.isOrReplace()) {
+        return new DdlCommandResult(true,
+            String.format("Cannot add table %s: A table with the same name "
+                    + "already exists.",
+                sourceName));
+      }
```

Comment:
Isn't this check now redundant, given there is code in `CreateSourceFactory` to do this now?  We should let the code do what it used to do, which is throw an exception, not return a success code!